### PR TITLE
test: Premium Course Certificates for Free Plan Users

### PR DESCRIPTION
**Pull request by acabreragnz**
### Overview
This PR introduces a series of new Gherkin features and Cypress tests covering the functionality of premium course certificates for free-plan users. It aims to ensure robust end-to-end testing of certificate visibility and accessibility based on the user's course completion status and device type.

### Changes
- Added Gherkin feature files for awarded and unawarded course certificates for free plan users.
- Implemented Cypress tests for scenarios including seeing "Download certificate" for awarded courses, the requirement to upgrade for unawarded courses, and device-specific advice for iOS users.

https://app.clickup.com/t/8687xxvrz

### CODE REVIEW

File: cypress/e2e/Courses/Certificates/FreePlanAwardedCourseCertificate.feature [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  Scenario: Free plan users should see the "Download certificate" option for completed and awarded premium courses
+    Given the user is not on an iOS mobile device
+    When the user selects the Completed tab
+    Then the user should see "Download certificate" for awarded courses
+
+  Scenario: Free plan users can successfully download certificates for awarded premium courses
+    Given the user is not on an iOS mobile device
+    And the user selects the Completed tab
+    And the user sees "Download certificate" for awarded courses
+    When the user clicks on "Download certificate"
```
[the remark this answers is not in the document]
Comment (gpmuch) [reply cypress/e2e/Courses/Certificates/FreePlanAwardedCourseCertificate.feature:15]: are we running different tests for different devices?

maybe we need to duplicate the feature files per device. I don't know what the standard is.

[the remark this answers is not in the document]
Comment (gpmuch) [reply cypress/e2e/Courses/Certificates/FreePlanAwardedCourseCertificate.feature:15]: If iOS specific, tap would be the correct verb.

File: cypress/e2e/Courses/Certificates/FreePlanAwardedCourseCertificate.feature [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    Then the certificate should be downloaded successfully
+
+  Scenario: Free plan users on iOS mobile devices should be advised to download certificates on desktop due to mobile restrictions
+    Given the user is on an iOS mobile device
+    When the user selects the Completed tab
+    Then the user should be advised to "Download the certificate from desktop"
+    And should not see the option to "Download certificate"
```
[the remark this answers is not in the document]
Comment (gpmuch) [reply cypress/e2e/Courses/Certificates/FreePlanAwardedCourseCertificate.feature:22]: "Call to Action"

File: cypress/e2e/Courses/Certificates/FreePlanAwardedCourseCertificate.feature [hunk 3]
```diff
@@ -0,0 +1,22 @@
+Feature: Awarded Premium Course Certificates for Free Plan Users
+  Background:
+    Given a free plan user has some completed awarded premium courses
+    And the user has navigated to the premium courses page
+
+  Scenario: Free plan users should see the "Download certificate" option for completed and awarded premium courses
+    Given the user is not on an iOS mobile device
+    When the user selects the Completed tab
```
[the remark this answers is not in the document]
Comment (acabreragnz) [reply cypress/e2e/Courses/Certificates/FreePlanAwardedCourseCertificate.feature:8]: Ok, we can keep it as it is.

After a `When` I prefer to write the noun of the person who is going to make the action as the first thing, 

```
Given the user does something
Given the system does something
Given I do something
```

https://cucumber.io/docs/gherkin/reference/#when

File: cypress/e2e/Courses/Certificates/FreePlanAwardedCourseCertificate.feature (hunk 2 above)
[the remark this answers is not in the document]
Comment (acabreragnz) [reply cypress/e2e/Courses/Certificates/FreePlanAwardedCourseCertificate.feature:22]: For this case I like that, but what if there are multiple non "call to action" buttons? (secondary button, etc)

File: cypress/e2e/Courses/Certificates/FreePlanAwardedCourseCertificate.feature (hunk 1 above)
[the remark this answers is not in the document]
Comment (acabreragnz) [reply cypress/e2e/Courses/Certificates/FreePlanAwardedCourseCertificate.feature:15]: instead of click/tap I will reword it to "the user request to ..."

[the remark this answers is not in the document]
Comment (acabreragnz) [reply cypress/e2e/Courses/Certificates/FreePlanAwardedCourseCertificate.feature:15]: > are we running different tests for different devices?

Only a few tests need mobile tests, for now, we are keeping them in the same files.

**Comment by acabreragnz** [comment]
@gpmuch I did some improvements in the wording of the gherking. Please review them async, I will merge this PR to have it as a base for the following tests.
